src/services/AdminApiContext.ts

```diff
@@ -123,31 +123,50 @@ export class AdminApiContext {
     }
 
     isAuthenticated(): boolean {
-        // TODO: check token expiry
-        return !!this.options['access_token'];
+        if (!this.options['access_token']) {
+            return false;
+        }
+
+        const token = this.options['access_token'];
+        const decodedToken: { exp: number } = jwtDecode(token);
```

Comment:
We should move the decode to the writing of the token option and save the expiry date in a separate option, so we don't have to decode the token on every request.

Just out of my head without testing, but something like this:

```JavaScript
// authenticateWithClientCredentials method

const authData = (await authResponse.json()) as { access_token?: string };

const decodedToken: { exp: number } = jwtDecode(authData['access_token']);

contextOptions['access_token_exp'] = decodedToken.exp;
```

```JavaScript
// isAuthenticated method

if (!this.options['access_token'] || !this.options['access_token_exp']) {
    return false;
}

return this.options['access_token_exp'] > Math.floor(Date.now() / 1000);
```